src/lib/eliom_client.client.ml

```diff
@@ -1253,19 +1287,23 @@ let () =
                 let rf = List.assq state_id.state_index !reload_functions in
                 reload_function := Some rf;
                 let%lwt () = run_lwt_callbacks ev (flush_onchangepage ()) in
-                current_state_id := state_id;
-                rf () () >>
-                (scroll_to_fragment ~offset:state.position fragment;
-                 Lwt.return_unit)
+                let old_page = !active_page in
+                (*TODO: too early*)
+                set_active_page @@ mk_page ~state_id ~status:Generating ();
+                let%lwt () = rf () () in
```

Comment:
Shouldn't we evaluate `rf () ()` within `Lwt.with_value this_page !active_page`?